src/ast/passes/codegen_llvm.h

```diff
@@ -257,6 +257,12 @@ class CodegenLLVM : public Visitor {
   void maybeAllocVariable(const std::string &var_ident,
                           const SizedType &var_type);
 
+  Function *DeclareKernelFunc(const std::string &func_name);
+
+  CallInst *CreateKernelFuncCall(const std::string &func_name,
```

Comment:
Can we do enums for kfuncs instead? Strings are easy to typo. Furthermore, kfuncs are unstable API. So I can see us ending up in a situation where we need `v1`, `v2`, ... `vN` variants for the same kfunc. Enums would help make that more manageable as well.

Reply:
Good idea, I'll give it a shot.

Reply:
Done, ready for another review.